android/app/src/main/java/org/fossasia/openevent/activities/SearchActivity.java

```diff
@@ -134,103 +124,33 @@ public void handleIntent(Intent intent) {
         searchQuery(query);
     }
 
-    private void searchQuery(String constraint) {
-        results.clear();
-
-        if (!TextUtils.isEmpty(constraint)) {
-            String query = constraint.toLowerCase(Locale.getDefault());
-            String wildcardQuery = String.format("*%s*", query);
-            addResultsFromTracks(wildcardQuery);
-            addResultFromSessions(wildcardQuery);
-            addResultsFromSpeakers(wildcardQuery);
-            addResultsFromLocations(wildcardQuery);
-        }
+    @Override
+    public void onSaveInstanceState(Bundle bundle) {
+        super.onSaveInstanceState(bundle);
     }
 
-    @Override
-    protected void onSaveInstanceState(Bundle outState) {
-        if (searchView != null) {
-            outState.putString(SEARCH, searchText);
-        }
-        super.onSaveInstanceState(outState);
+    private void searchQuery(String constraint) {
+        searchViewModel.getSearchResults(constraint).observe(SearchActivity.this, searches -> {
+            results.clear();
+            results.add(searches);
+            globalSearchAdapter.setCopyOfSearches(searches);
+            globalSearchAdapter.notifyDataSetChanged();
+            handleVisibility();
+        });
     }
 
     @Override
     public void onDestroy() {
         super.onDestroy();
         OpenEventApp.getEventBus().unregister(this);
-
         //Closing realm instance and detaching listeners to avoid memory leaks
         realm.close();
         realm.removeAllChangeListeners();
```

Comment:
Why are there realm instances here now? Every database interaction should be moved to ViewModel